tests/test_flask_util.py

```diff
@@ -0,0 +1,325 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Unit tests for the Flask utilities"""
+
+__author__ = 'jonwayne@google.com (Jon Wayne Parrott)'
+
+import httplib2
+import json
+import unittest
+
+import flask
+import mock
+
+from oauth2client import GOOGLE_AUTH_URI
+from oauth2client import GOOGLE_TOKEN_URI
+from oauth2client import clientsecrets
+from oauth2client.flask_util import OAuth2 as FlaskOAuth2
+from oauth2client.client import OAuth2Credentials
+
+
+class Http2Mock(object):
+    """Mock httplib2.Http for code exchange / refresh"""
+    def __init__(self, status=200, **kwargs):
+        self.status = status
+        self.content = {
+            'access_token': 'foo_access_token',
+            'refresh_token': 'foo_refresh_token',
+            'expires_in': 3600,
+            'extra': 'value',
+        }
+        self.content.update(kwargs)
+
+    def request(self, token_uri, method, body, headers, *args, **kwargs):
+        self.body = body
+        self.headers = headers
+        return (self, json.dumps(self.content).encode('utf-8'))
+
+    def __enter__(self):
+        self.httplib2_orig = httplib2.Http
+        httplib2.Http = self
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        httplib2.Http = self.httplib2_orig
+
+    def __call__(self, *args, **kwargs):
+        return self
+
+
+class FlaskOAuth2Tests(unittest.TestCase):
+
+    def setUp(self):
+        self.app = flask.Flask(__name__)
+        self.app.testing = True
+        self.app.config['SECRET_KEY'] = 'notasecert'
+        self.oauth2 = FlaskOAuth2(
+            self.app,
+            client_id='client_idz',
+            client_secret='client_secretz')
+
+    def _generate_credentials(self):
+        return OAuth2Credentials(
+            'access_tokenz',
+            'client_idz',
+            'client_secretz',
+            'refresh_tokenz',
+            '3600',
+            GOOGLE_TOKEN_URI,
+            'Test',
+            id_token={
+                'sub': '123',
+                'email': 'user@example.com'
+            })
+
+    def test_explicit_configuration(self):
+        oauth2 = FlaskOAuth2(
+            flask.Flask(__name__), client_id='id', client_secret='secret')
+
+        assert oauth2.client_id == 'id'
```

Comment:
Do you want me to replace all the standalone assertions with their method equivalents?